spec.emu

```diff
@@ -0,0 +1,108 @@
+<pre class="metadata">
+title: JavaScript Module Blocks
+stage: 1
+contributors: Surma, Dan Ehrenberg
+</pre>
+
+<emu-clause id="sec-primary-expression">
+  <h1>Primary Expression</h1>
+  <h2>Syntax</h2>
+  <emu-grammar type="definition">
+    PrimaryExpression[Yield, Await] :
+      IdentifierReference[?Yield, ?Await]
+      Literal
+      ArrayLiteral[?Yield, ?Await]
+      ObjectLiteral[?Yield, ?Await]
+      FunctionExpression
+      ClassExpression[?Yield, ?Await]
+      GeneratorExpression
+      AsyncFunctionExpression
+      AsyncGeneratorExpression
+      <ins>ModuleBlockExpression</ins>
+      RegularExpressionLiteral
+      TemplateLiteral[?Yield, ?Await, ~Tagged]
+      CoverParenthesizedExpressionAndArrowParameterList[?Yield, ?Await] #parencover
+  </emu-grammar>
+</emu-clause>
+
+<emu-clause id="sec-module-block-definitions">
+  <h1>Module Block Definitions</h1>
+  <h2>Syntax</h2>
+  <emu-grammar type="definition">
+    ModuleBlockExpression :
+      `module` [no LineTerminator here] `{` ModuleBody_opt? `}`
+  </emu-grammar>
+
+  <h2>Runtime Semantics: Evaluation</h2>
+  <emu-alg>
+    1. Let _moduleBlock_ be OrdinaryObjectCreate(%ModuleBlock.prototype%,  &laquo; [[HostDefinedSlot]], [[ModuleBlockBody]], [[SourceText]] &raquo;).
+    1. Set _moduleBlock_.[[SourceText]] to the source text matched by |ModuleBlockExpression|.
+    1. Perform ! HostInitializeModuleBlock(_moduleBlock_).
+    1. Return _moduleBlock_.
+  </emu-alg>
+</emu-clause>
+  
+<emu-clause id="sec-moduleblock">
+  <h1>ModuleBlock ( _body_ )</h1>
+  <p>The _body_ specifies the body (executable code) of a module block.</p>
+  <p>When the `ModuleBlock` function is called,the following steps are taken:</p>
+  <emu-alg>
+    1. If NewTarget is *undefined*
+      1. Throw a *TypeError* exception.
```

Comment:
We should call https://tc39.es/ecma262/#sec-hostensurecancompilestrings before ParseText

Reply:
But this is not code generation from a string. Why it should call this hook?

Reply:
In what sense is it not code generation from a string?

Reply:
eval and Function are code generation but module block not

Reply:
Well, another choice would be to do the check the first time it is imported. I guess this would move it to the host. Is this the interpretation you prefer?

Reply:
Oh, wait so this is a new constructor for module block?

```js
const a = module { expr }
```

Equals to this?

```js
const ctor = GetConstructor(module {})
const x = new ctor('expr')
```